### PR TITLE
[2031] Workflows for database backup/restore

### PR DESCRIPTION
**Pull request by saliceti**
### Context
Add workflows for disaster recovery situations

### Changes proposed in this pull request
- Refactor the backup workflow to use the standard github action
- Add Point in time restore workflow
- Add restore from storage workflow

### Guidance to review
Code review. Workflows must be merged to main to be able to test the workflow dispatch trigger

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [x] Cleaned commit history
-   [ ] Tested by running locally

### CODE REVIEW

**Comment by saliceti** [comment]
[TRS DR postgres workflow setup](https://trello.com/c/jmodSQ3K/2031-trs-dr-postgres-workflow-setup)

File: .github/workflows/backup-db.yml [hunk 1]
```diff
@@ -2,71 +2,92 @@ name: Backup production database
...
+      run: |
+        echo "PTR_DB_SERVER=${{ env.DB_SERVER }}-ptr" >> $GITHUB_ENV
+        echo "BACKUP_FILE=${{ env.BACKUP_FILE }}-ptr" >> $GITHUB_ENV
+
+    - name: Backup ${{ env.DEPLOY_ENV }} postgres
+      uses: DFE-Digital/github-actions/backup-postgres@master
+      with:
+        storage-account: ${{ env.STORAGE_ACCOUNT_NAME }}
+        resource-group: ${{ env.RESOURCE_GROUP_NAME }}
+        app-name: ${{ env.SERVICE_NAME }}-${{ env.DEPLOY_ENV }}
```
Comment (RMcVelia) [inline .github/workflows/backup-db.yml:75]: I believe the app name will need to be -ui or -api

File: .github/workflows/postgres-restore.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        options:
+        - 'false'
+        - 'true'
+      backup-file:
+        description: Name of the backup file in Azure storage. e.g. trs_prod_2024-08-09.tar.gz. The default value is today's scheduled backup.
+        type: string
+        required: false
+
+env:
+  SERVICE_NAME: itt-mentor-services
```
Comment (RMcVelia) [inline .github/workflows/postgres-restore.yml:28]: trs?

File: .github/workflows/postgres-restore.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  SERVICE_SHORT: ittms
+  TF_VARS_PATH: terraform/application/config
```
Comment (RMcVelia) [inline .github/workflows/postgres-restore.yml:30]: terraform/aks/config

**Review by RMcVelia — APPROVED** [review]

**Pull request merged**